api/repositories/droplet_repository.go

```diff
@@ -87,6 +87,40 @@ func (r *DropletRepo) GetDroplet(ctx context.Context, authInfo authorization.Inf
 	return cfBuildToDroplet(build)
 }
 
+func (r *DropletRepo) GetDroplets(ctx context.Context, authInfo authorization.Info, appGuid string) ([]DropletRecord, error) {
+	buildList := &korifiv1alpha1.CFBuildList{}
+
+	namespaces, err := r.namespacePermissions.GetAuthorizedSpaceNamespaces(ctx, authInfo)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list namespaces for spaces with user role bindings: %w", err)
+	}
+
+	userClient, err := r.userClientFactory.BuildClient(authInfo)
+	if err != nil {
+		return []DropletRecord{}, fmt.Errorf("failed to build user client: %w", err)
+	}
+
+	var allBuilds []korifiv1alpha1.CFBuild
+	for ns := range namespaces {
+		err := userClient.List(ctx, buildList, client.InNamespace(ns))
```

Comment:
The [cfbuild webhook](https://github.com/cloudfoundry/korifi/blob/d0483c8b89cc8112f731f882a8dfe1485f0b6cb5/controllers/api/v1alpha1/cfbuild_webhook.go#L45) sets the `korifi.cloudfoundry.org/app-guid` label on the CFBuild to the app guid. Therefore, when listing builds you could leverage label selectors. Then the additional filtering on builds below would not be necessary.

Here is an [example](https://github.com/cloudfoundry/korifi/blob/d0483c8b89cc8112f731f882a8dfe1485f0b6cb5/controllers/controllers/workloads/processes/controller.go#L89) how to use label selectors when listing